### PR TITLE
Refactoring UI Adapter and CLI

### PR DESCRIPTION
**Pull request by avincigu**
Refactored UI Adapter to NRE code style. Also added a significant amount of unit tests. Deleted unused code.

### CODE REVIEW

File: cli/ucs_completion.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ _ucs_completions()
     commands="group view"
     subcommands=(
                  "add remove get list"
-                 "system-info event env inventory-history replacement-history inventory-info state network-config snapshot-info snapshot-getref jobpower"
+                 "system-info event env state network-config job reservation"
```
Comment (yhuan47) [inline cli/ucs_completion.sh]: Are inventory related commands totally removed? This is still required for NRE.

Comment (avincigu) [reply cli/ucs_completion.sh]: They were commented on the previous version, so I removed it. But if we determine that we want them in nre I can add them back in. You are refering to both inventory-info and inventory-history?

File: cli/cli/src/json_display.py [hunk 2]
```diff
@@ -5,32 +5,16 @@
...
-                 "K": "Kernel boot started",
-                 "A": "Active",
-                 "M": "Missing",
-                 "E": "Error",
-                 "U": "Unkwown"}
-    WLMSTATE_MAP = {"A": "Available",
-                    "U": "Unavailable",
-                    "M": "Maintenance",
-                    "X": "Unknown"}
-
```
Comment (paul-amonson) [inline cli/cli/src/json_display.py]: Is this not needed anymore?

File: cli/ucs_completion.sh (hunk 1 above)
Comment (paul-amonson) [reply cli/ucs_completion.sh]: PO inventory is currently != NRE Inventory. I would suggest removing them and then re-adding them when inventory in NRE is completed. There will be different and all new queries.

File: cli/cli/src/json_display.py (hunk 2 above)
Comment (avincigu) [reply cli/cli/src/json_display.py]: Yes, this mapping happens in the UI Adapter now.

Comment (paul-amonson) [reply cli/cli/src/json_display.py]: Great!

File: cli/ucs_completion.sh (hunk 1 above)
Comment (yhuan47) [reply cli/ucs_completion.sh]: Make sense, please let Ming involve in the new commands.

**Pull request merged**